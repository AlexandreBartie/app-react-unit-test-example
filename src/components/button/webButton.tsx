import React from 'react';
import './webButton.css'

type IButton = { label?: string };

export function WebButton(props: IButton) {
  return (
    <div data-testid="button" className="button-style">
      {props.label}
    </div>
  );
}
