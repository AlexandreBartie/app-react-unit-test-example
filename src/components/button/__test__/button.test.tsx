import React from 'react';
import { render, screen } from '@testing-library/react';
import { WebButton } from '../webButton';

it('check render', async () => {

    render(<WebButton/>);

    const linkElement = screen.logTestingPlaygroundURL

    expect(linkElement).not.toBeNull()

})

it('check label', async () => {

    const tag = 'Hello'

    render(<WebButton label={tag}></WebButton>);

    const linkElement = await screen.findByText(tag);

    expect(linkElement).not.toBeNull()

})

it('check testId', async () => {

    render(<WebButton label='Hello'></WebButton>);

    const linkElement = await screen.findByTestId('button');

    expect(linkElement).not.toBeNull()

});


