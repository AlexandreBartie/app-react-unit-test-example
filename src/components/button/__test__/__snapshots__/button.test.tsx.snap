// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check master 1`] = `
<div
  className="button-style"
  data-testid="button"
>
  HMasterello
</div>
`;

exports[`check master 2`] = `
<div
  className="button-style"
  data-testid="button"
>
  Deise
</div>
`;

exports[`check tree 1`] = `
<div
  className="button-style"
  data-testid="button"
>
  HMasterello
</div>
`;
